fuzz/fuzz_targets/encode_decode.rs

```diff
@@ -28,11 +41,13 @@ fn do_test(data: &[u8]) {
         bech32::Variant::Bech32
     };
 
-    if let Ok(data_str) = bech32::encode(&hrp, &dp, variant).map(|b32| b32.to_string()) {
-        let decoded = bech32::decode(&data_str);
-        let b32 = decoded.expect("should be able to decode own encoding");
+    if let Ok(hrp) = Hrp::parse(&hrp) {
+        if let Ok(encoded) = bech32::encode(hrp, &dp, variant).map(|b32| b32.to_string()) {
```

Comment:
In 599ce6d089904386ac6a34d77a6b3b1fe222393f:

What is the purpose of this `.to_string`?

Reply:
derp, what was I even mapping.